packages/inventory/src/services/inventory-item.ts

```diff
@@ -198,38 +207,44 @@ export default class InventoryItemService {
       "id" | "created_at" | "metadata" | "deleted_at"
     >
   ): Promise<InventoryItem> {
-    const manager = this.getManager()
-    const itemRepository = manager.getRepository(InventoryItem)
+    return await this.atomicPhase_(async (manager) => {
+      const itemRepository = manager.getRepository(InventoryItem)
 
-    const item = await this.retrieve(inventoryItemId)
+      const item = await this.retrieve(inventoryItemId)
 
-    const shouldUpdate = Object.keys(data).some((key) => {
-      return item[key] !== data[key]
-    })
+      const shouldUpdate = Object.keys(data).some((key) => {
+        return item[key] !== data[key]
+      })
 
-    if (shouldUpdate) {
-      itemRepository.merge(item, data)
-      await itemRepository.save(item)
+      if (shouldUpdate) {
+        itemRepository.merge(item, data)
+        await itemRepository.save(item)
 
-      await this.eventBusService_.emit(InventoryItemService.Events.UPDATED, {
-        id: item.id,
-      })
-    }
+        await this.eventBusService_
+          .withTransaction(manager)
```

Comment:
thought: if the module is not registered as a shared module this would break. The reason is the following:

- core manager creates transaction
- `inventoryService.withTransaction(coreManager).something()`
- now InventoryService is trying to use a manager that doesn't have the models from the module registered so can't do e.g. `manager.getRepository(InventoryItem)`

Alternatively, if the core doesn't call the inventoryService.withTransaction the following happens:

- InventoryService calls atomicPhase which creates a new transaction using the InventoryService connection.
- the transactionManager created by atomicPhase doesn't know about the StagedJob model so calling `eventBusService.withTransaction(inventoryManager)` will fail.

Again I think it might make sense, the first time around to only support the shared approach and not leave any code paths open that cannot work.

Reply:
A thing to be aware of in any case is that when we start looking at distributing these this would be a thing that would need refactoring of some sort.

Reply:
Good catch. Just refactored it a bit and tested the flow manually to check.
I'm restricting the modules to be only be used with shared resources.
When EventBus become a Module we refactor modules using it.